### ISSUE TITLE
core:  CfnResource.isCfnResource is too permissive

### ISSUE THREAD

**Issue by michanto** [aws/aws-cdk#30473]
### Describe the bug

I created a construct class with a cfnResourceType.  I use that type to search for a CfnResource under the parent of that construct.  But my non-CfnResource class is recognized as a CfnResource by the CDK because it has a cfnResourceType property!  Wrong!

### Expected Behavior

```
public static isCfnResource(x: any): x is CfnResource {
    return CfnElement.isCfnElement(x) && x.cfnResourceType !== undefined;
  }
```

That would be a lot better because isCfnElement has an RTTI symbol property.

### Current Behavior

```
public static isCfnResource(x: any): x is CfnResource {
    return x !== null && typeof(x) === 'object' && x.cfnResourceType !== undefined;
  }
```

### Reproduction Steps

```
class BucketHelperClass extends Construct {
  readonly cfnResourceType = CfnBucket.CFN_RESOURCE_TYPE_NAME
}
let helper = new BucketHelperClass(new App(), "Helper")
if (CfnResource.isCfnResource(helper)) {
  throw new Error('No it is not a CfnResource!');
}
```

### Possible Solution

Either this:
```
public static isCfnResource(x: any): x is CfnResource {
    return CfnElement.isCfnElement(x) && x.cfnResourceType !== undefined;
  }
```

Or maybe:

```
const CFN_RESOURCE_SYMBOL = Symbol.for('@aws-cdk/core.CfnResource');
```

### CDK CLI Version

2.120.0 (build 58b90c4)

### Node.js Version

10.2.5

### OS

MacOS

### Language

TypeScript

**Comment by pahud** [comment]
Agree. But looking at this discussion

https://github.com/aws/aws-cdk/pull/28001#pullrequestreview-1731529704

What do you think about it? @tmokmss @go-to-k  ?

**Comment by michanto** [comment]
Both could be supported by just adding isCfnElement to the existing?

**Comment by tmokmss** [comment]
I agree that the current state is undesirable and should be fixed. However, I can't think of a way to make sure it's not a breaking change. I think the proposed change will not cause any problem, but there are so many cdk users in the world...

**Comment by go-to-k** [comment]
I have the same opinion with @tmokmss . (But I think it's probably OK.)

**Comment by michanto** [comment]
Even if you decide not to fix isCfnResource itself, you should fix the addDependency error on synthesis.  That's definitely a bug.
